review/developer/before-submitting-prs.md

```diff
@@ -0,0 +1,23 @@
+# Before Submitting PRs
+
+Here are points to check out before submitting a PR
+
+## Code
+- [ ] Code follows the
+  - Guide in of this project
+  - [Developer Guides](https://github.com/tyemill/developer-guides)
+  - [Ruby Guides](https://github.com/rubocop/ruby-style-guide)
+  - [Rails Guide](https://github.com/rubocop/rails-style-guide)
+
+- [ ] Performed a self-review of the code
+- [ ] Added tests that prove the fix is effective or that the feature works
+- [ ] New and existing unit tests pass locally with new changes
+- [ ] Code is commented, particularly in hard-to-understand areas
+- [ ] Made corresponding changes to the documentation. eg. README.md, Github Wiki
+- [ ] Changes generate no new warnings
+- [ ] Verified that the fix/feature does not cause performance issues
+
+## UI (Choose one of the following):
+- [ ] Verified styling and correct element sizing, and have tested at our minimum application width
+- [ ] There is a separate task for UI (link to task)
+- [ ] This task is purely back-end
```

Comment:
I will like Lead Devs to review this content to ensure it match dev team guide.

Reply:
I'd be nice to add these to other projects's pull_request_template. It works really good for Lexicon team
https://github.com/tyemill/lexicon/blob/SPRINT/.github/pull_request_template.md?plain=1

Reply:
We did talked about in Syntax retro, people already follows the guide, they will prefer to add https://github.com/tyemill/code-review-practices, instead of in a pull request template, because they feel it is developer's responsibility to get a PR be ready to review.

Then a checklist will be in https://github.com/tyemill/code-review-practices to let developers to follow best practice.

Reply:
the nice thing about having it is to not forget to cover all these points (reflected later on verification) and enforce better process but I can see people referring to this as well 👍